webapp-django/crashstats/crashstats/models.py

```diff
@@ -1709,6 +1709,62 @@ class GraphicsDevices(SocorroMiddleware):
     def post(self, payload):
         return super(GraphicsDevices, self).post(self.URL_PREFIX, payload)
 
+    def get_pairs(self, adapter_hexes, vendor_hexes):
+        """return a dict where each tuple of (adapter_hex, vendor_hex)
+        corresponds to a (adapter_name, vendor_name) pair."""
+        assert len(adapter_hexes) == len(vendor_hexes)
+        names = {}
+        missing = {}
+        for i, adapter_hex in enumerate(adapter_hexes):
+            cache_key = (
+                'graphics_adapters' + adapter_hex + ':' + vendor_hexes[i]
+            )
+            name_pair = cache.get(cache_key)
+            key = (adapter_hex, vendor_hexes[i])
+            if name_pair is not None:
+                names[key] = name_pair
+            else:
+                missing[key] = cache_key
+
+        missing_vendor_hexes = []
+        missing_adapter_hexes = []
+        for adapter_hex, vendor_hex in missing:
+            missing_adapter_hexes.append(adapter_hex)
+            missing_vendor_hexes.append(vendor_hex)
```

Comment:
```python
missing_vendor_hexes = [x[1] for x in missing]
missing_adapter_hexes = [x[0] for x in missing]
```

Isn't that the same thing, more or less? I wonder if there's any impact in doing the loop twice instead of just once.